### PR TITLE
Add Chromium data for WebGPU supported features

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

This PR adds compat data for all the [additional features](https://gpuweb.github.io/gpuweb/#feature-index) defined in the WebGPU spec that can be enabled for a `GPUDevice`.

It also adds data for API features that are available when certain additional features are enabled.

Data sources:

- clip-distances: https://chromestatus.com/feature/5141929256550400
- timestamp-queries: https://developer.chrome.com/blog/new-in-webgpu-121#timestamp_queries_in_compute_and_render_passes
- dual-source-blending: https://developer.chrome.com/blog/new-in-webgpu-130#dual_source_blending
- float32-filterable: https://developer.chrome.com/blog/new-in-webgpu-119#filterable_32-bit_float_textures
- shader-f16: https://developer.chrome.com/blog/new-in-webgpu-120#support_for_16-bit_floating-point_values_in_wgsl
- float32-blendable: https://chromestatus.com/feature/5173655901044736

For features that I couldn't find specific data on, I am currently just assuming that they were supported at the same time as WebGPU first shipped (Chrome 113, Chrome Android 121), but this may not be true for all of them. I will check these with the Chrome eng folk, going forward.

#### Related issues

Part of the [Bring MDN up-to-date with WebGPU updates](https://github.com/mdn/mdn/issues/591) project.

### CODE REVIEW

File: api/GPUSupportedFeatures.json [hunk 1]
```diff
@@ -113,6 +113,825 @@
...
+        }
+      },
+      "feature_texture-compression-astc-sliced-3d": {
+        "__compat": {
+          "description": "<code>texture-compression-astc-sliced-3d</code> feature",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#texture-compression-astc-sliced-3d",
+          "tags": [
+            "web-features:webgpu"
+          ],
+          "support": {
```
Comment (beaufortfrancois) [inline api/GPUSupportedFeatures.json:681]: It's not supported in Chrome yet. Only in the spec.

File: api/GPUSupportedFeatures.json [hunk 2]
```diff
@@ -113,6 +113,825 @@
...
+        }
+      },
+      "feature_texture-compression-bc-sliced-3d": {
+        "__compat": {
+          "description": "<code>texture-compression-bc-sliced-3d</code> feature",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#texture-compression-bc-sliced-3d",
+          "tags": [
+            "web-features:webgpu"
+          ],
+          "support": {
```
Comment (beaufortfrancois) [inline api/GPUSupportedFeatures.json:791]: It's not supported in Chrome yet.

File: api/GPUSupportedFeatures.json (hunk 1 above)
Comment (chrisdavidmills) [reply api/GPUSupportedFeatures.json]: Ah, OK. BCD doesn't let you include features not supported anywhere, so I've deleted this data point for now. I'll add notes to the corresponding content PR to make it clear that these currently aren't supported anywhere.

File: api/GPUSupportedFeatures.json (hunk 2 above)
Comment (chrisdavidmills) [reply api/GPUSupportedFeatures.json]: Same comment as above.

**Comment by chrisdavidmills** [comment]
@Elchi3 hello mate, I was wondering if you could help me with this one? The linter is failing because of "errors" related to MDN URLs not being added to data points completely unrelated to my changes. And I casn't seem to fix them either, because those errors do not appear on my local branch when I run `npm run test`, even though it is up-to-date with the remote branch.

**Comment by Elchi3** [comment]
Hey Chris, seems like you introduced the mdn_url removals in https://github.com/mdn/browser-compat-data/pull/24760/commits/750f25732d96d2315bf30ba825bfb64f7fdb3e36. 

Can you: pull from this branch (I just merged main), run `npm i` to install the latest dependencies, run `npm run fix` to fix the mdn_urls, and commit and push that back into here?

**Comment by chrisdavidmills** [comment]
> you: pull from this branch (I just merged main), run `npm i` to install the latest dependencies, run `npm run fix` to fix the mdn_urls, and commit and push that back into here?

@Elchi3 Done! Thanks for helping me figure this out.

File: api/GPUSupportedFeatures.json [hunk 3]
```diff
@@ -113,6 +113,715 @@
...
+                    "type": "runtime_flag",
+                    "name": "--unstable-webgpu"
+                  }
+                ]
+              },
+              {
+                "version_added": "1.8",
+                "version_removed": "1.32"
+              }
+            ],
```
Comment (caugner) [inline api/GPUSupportedFeatures.json:144]: How did you determine the Deno support status for the new `GPUSupportedFeatures` subfeatures?

File: api/GPUCommandEncoder.json [hunk 4]
```diff
@@ -117,6 +117,50 @@
...
+              "firefox_android": "mirror",
+              "ie": {
+                "version_added": false
+              },
+              "oculus": "mirror",
+              "opera": "mirror",
+              "opera_android": "mirror",
+              "safari": {
+                "version_added": "preview"
+              },
```
Comment (caugner) [inline api/GPUCommandEncoder.json:152]: How did you determine the Safari TP support status of all the newly added features?

**Review by caugner — COMMENTED** [review]
Overall, the Chromium-related changes seem plausible, although some features like `feature_rg11b10ufloat-renderable` don't seem to have a source cited in the PR description.

Otherwise, I have two provenance questions about the Deno and Safari TP support statements. The fact that all those sub-features haven't all landed in the same Chromium versions suggests to me that it is unlikely that Deno and Safari TP started supporting all these in the same version as the parent feature.

@chrisdavidmills To facilitate the review of this PR, could you consider starting a thread on the line of each feature's name that states the data provenance for Chrome, Deno, and Safari. For Chrome, this could be a link to the corresponding "What's New in WebGPU" blog post and/or a short statement about the reasoning.

**Comment by chrisdavidmills** [comment]
@caugner including the untested deno and safari data was an oversight. I've now set all those values to `false`.

Proof of Chrome support for all the features is included at the top of the thread, and this PR has been reviewed by Chrome engineering.

**Comment by chrisdavidmills** [comment]
> > Proof of Chrome support for all the features is included at the top of the thread, and this PR has been reviewed by Chrome engineering.
> 
> I just checked again, and none of the referenced links mention `bgra8unorm`.
> 
> My main point is that as PR author you can make reviewing a big PR like this a lot easier (and reduce time to merge) if you provide sources explicitly for each feature. It's significantly more work for me as reviewer to manually match each feature against the sources provided (possibly more than once if the review leads to partial rework of the PR).

`bgra8unorm` and other features not explicitly mentioned in the provided data links are covered by the paragraph that follows the list of links:

> For features that I couldn't find specific data on, I am currently just assuming that they were supported at the same time as WebGPU first shipped (Chrome 113, Chrome Android 121), but this may not be true for all of them. I will check these with the Chrome eng folk, going forward.

I couldn't find any explicit data sources for those, so I instead deferred to Chrome engineering. François' review verified the data I'd added for those (or in a couple of cases, gave me the correct data where I'd gone wrong).

**Review by caugner — APPROVED** [review]

**Pull request merged**